src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs

```diff
@@ -383,5 +391,101 @@ public IEnumerable<ValidationResult> Validate(
                 return validationResults;
             }
         }
+
+        internal class AllowedTypeValidator : ITypedJsonValidator<List<MediaWithCropsDto>, MediaPicker3Configuration>
+        {
+            private readonly ILocalizedTextService _localizedTextService;
+            private readonly IMediaTypeService _mediaTypeService;
+
+            public AllowedTypeValidator(ILocalizedTextService localizedTextService, IMediaTypeService mediaTypeService)
+            {
+                _localizedTextService = localizedTextService;
+                _mediaTypeService = mediaTypeService;
+            }
+
+
+            public IEnumerable<ValidationResult> Validate(
+                List<MediaWithCropsDto>? value,
+                MediaPicker3Configuration? configuration,
+                string? valueType,
+                PropertyValidationContext validationContext)
+            {
+                if (value is null || configuration is null)
+                {
+                    return [];
+                }
+
+                var allowedTypes = configuration.Filter?.Split(Constants.CharArrays.Comma, StringSplitOptions.RemoveEmptyEntries);
+
+                // No allowed types = all types are allowed
+                if (allowedTypes is null || allowedTypes.Length == 0)
+                {
+                    return [];
+                }
+
+                foreach (MediaWithCropsDto media in value)
```

Comment:
Looks like you could de-duplicate here to avoid multiple checks that we already know the result of.  I.e. if you had three media of the same type here, you could first find all the distinct media type aliases and loop through those, and hence only need to do one look up of the media type and one check.